### PR TITLE
Refactored Employee Service

### PR DESCRIPTION
**Pull request by madwizzard**
Screenshot 2024-07-26 105138
Screenshot 2024-07-26 105150

The email sending an errorlog test needs to be worked on, but the rest of the lines are covered.

### CODE REVIEW

**Review by Juanro-Minne — APPROVED** [review]

File: HRIS.Services/Services/EmployeeService.cs [hunk 1]
```diff
@@ -122,45 +128,54 @@ public async Task<List<EmployeeDto>> GetAll(string userEmail = "")
 
     public async Task<EmployeeDto?> GetEmployee(string email)
```
Comment (EduanO) [inline HRIS.Services/Services/EmployeeService.cs]:
```suggestion
    public async Task<EmployeeDto?> GetEmployeeByEmail(string email)
```

Naming of methods is important to describe them better

**Review by EduanO — CHANGES_REQUESTED** [review]

File: HRIS.Services/Interfaces/IEmployeeService.cs
Comment (EduanO) [inline HRIS.Services/Interfaces/IEmployeeService.cs:1]: We do not need `GetEmployeeById` and `GetById` As they do the exact same thing except the one does an include and the otherr does not.

**Review by EduanO — CHANGES_REQUESTED** [review]

File: HRIS.Services/Services/EmployeeService.cs [hunk 2]
```diff
@@ -122,45 +128,54 @@ public async Task<List<EmployeeDto>> GetAll(string userEmail = "")
...
                                 .Include(employee => employee.EmployeeType)
                                 .Include(employee => employee.PhysicalAddress)
                                 .Include(employee => employee.PostalAddress)
                                 .Select(employee => employee.ToDto())
                                 .FirstOrDefaultAsync() ?? throw new CustomException("Unable to Load Employee");
+
+        return employee;
     }
 
     public async Task<EmployeeDto> UpdateEmployee(EmployeeDto employeeDto, string email)
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeService.cs:167]: Logic: is string email actually being used in the service, controller and front end or are we using employeeDto.email?

File: HRIS.Services/Services/EmployeeService.cs [hunk 3]
```diff
@@ -122,45 +128,54 @@ public async Task<List<EmployeeDto>> GetAll(string userEmail = "")
...
     {
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeService.cs]: Logic: is string email actually being used in the service, controller and front end or are we using employeeDto.email?

**Review by EduanO — APPROVED** [review]

**Review by DewanAlberts — APPROVED** [review]

**Pull request merged**